### PR TITLE
Wei Techdebt Add True-to-Usage Testcase to Cappa's Testcases

### PR DESCRIPTION
**Pull request by WeiYou93**
<PR Name> Add True-to-Usage Testcase to Cappa's Testcases
-----------------------------------------
For @ChristopherSchuster 

<High level description of feature> 
Add test cases for cappa install that asserts on the presence and package version for:
bower
npm
npmg
pip
system
tsd
captricity(internal-api-clients)

Tests to run:
```python setup.py test```

REFERENCE
- <https://captricity.atlassian.net/browse/ENG-4620> JIRA ticket

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def setUp(self):
         self._setup_cappa()
 
     def tearDown(self):
-        self.vagrant.destroy()
+        # self.vagrant.destroy()
```
Comment (ChristopherSchuster) [inline tests/base.py]: Shouldn't be commented.

**Comment by WeiYou93** [comment]
Just uncommented.

File: tests/test_bower.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            "**/.*",
+            "node_modules",
+            "bower_components",
+            "test",
+            "tests"
+        ]
+    }
+}"""
+
+TEST_NECESSARY_PROVISIONERS = """{
```
Comment (ChristopherSchuster) [inline tests/test_bower.py]: I'd move these constants to a common file and given them more descriptive names so we don't have the same constants defined in many test cases.

File: tests/base.py [hunk 3]
```diff
@@ -22,10 +22,25 @@ def requirements_json_install_factory(requirements_json):
...
     return requirements_json_install
 
+def requirements_json_install_factory_virtualenv(requirements_json):
+    """Given the contents of the requirements.json file, this will return a
+    fabric task that will create the file on the remote machine and install the
+    requirements using cappa.
+    """
+
+    @task
+    def requirements_json_install_virtualenv():
```
Comment (ChristopherSchuster) [inline tests/base.py]: with virtualenv is the default.

File: tests/test_captricity.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        return install_requirements_json_captricity
+
+
+TEST_INSTALL_CAPTRICITY_VERSION_JSON = """{
+    "Captricity": {
+        "pip": ["internal-api-clients@v0.0.3"]
+    }
+}"""
+
+TEST_NECESSARY_PROVISIONERS = """{
```
Comment (ChristopherSchuster) [inline tests/test_captricity.py]: Same deal with constants here

File: tests/test_bower.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        "dependencies": {
+            "jquery": "2.1.4",
+            "lodash": "3.9.3",
+            "angular": "1.4.0",
+            "angular-animate": "1.4.0",
+            "angular-sanitize": "1.4.0",
+            "angular-cookies": "1.4.0",
+            "angular-strap": "2.2.4",
+            "angular-mocks": "1.4.0",
+            "bootstrap": "3.3.5",
```
Comment (ChristopherSchuster) [inline tests/test_bower.py]: This is the only package being asserted on.

File: tests/test_tsd.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+class TsdTestCases(VagrantTestCase):
+
+    def test_basic(self):
+        self.install_requirements_json(NPM_DEPENDENCIES)
+        self.install_requirements_json(TSD_DEPENDENCIES)
+        self.install_requirements_json(TEST_TSD_VERSION_JASON)
+        self.run_spec('tsd_install_basic_spec')
+
+
+TEST_TSD_VERSION_JASON = """{
```
Comment (ChristopherSchuster) [inline tests/test_tsd.py]: Jason?

**Comment by ChristopherSchuster** [comment]
LGTM.

**Pull request merged**